lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -520,7 +520,12 @@ def request_headers(options)
         headers
       end
 
-      def success_from(action, response)
+      def success_from(action, response, options)
+        if response.dig('resultCode') == 'RedirectShopper' && !options[:execute_threed] && !options[:threed_dynamic]
```

Comment:
I know it's possible sometimes to have a direct challenge that skips the `RedirectShooper` result code and goes straight to `ChallengeShopper` - would that be possible in the dynamic 3ds scenario and do we need to check for that as well?

Reply:
Great question, I'm not sure if it's possible but it seems like a good idea to add it to the criteria.